src/Umbraco.Infrastructure/Migrations/Upgrade/UmbracoPlan.cs

```diff
@@ -1,289 +1,56 @@
-
-                // only from 8.0.0 and above
-                var minVersion = new SemVersion(8);
-                if (currentVersion < minVersion)
-                {
-                    throw new InvalidOperationException(
-                        $"Version {currentVersion} cannot be migrated to {_umbracoVersion.SemanticVersion}."
-                        + $" Please upgrade first to at least {minVersion}.");
-                }
-
-                // Force versions between 7.14.*-7.15.* into into 7.14 initial state. Because there is no db-changes,
-                // and we don't want users to workaround my putting in version 7.14.0 them self.
-                if (minVersion <= currentVersion && currentVersion < new SemVersion(7, 16))
-                {
-                    return GetInitState(minVersion);
-                }
-
-                // initial state is eg "{init-7.14.0}"
-                return GetInitState(currentVersion);
-            }
-        }
-
-        /// <summary>
-        ///     Gets the initial state corresponding to a version.
-        /// </summary>
-        /// <param name="version">The version.</param>
-        /// <returns>
-        ///     The initial state.
-        /// </returns>
-        private static string GetInitState(SemVersion version) => InitPrefix + version + InitSuffix;
+            => DefinePlan();
 
         /// <summary>
-        ///     Tries to extract a version from an initial state.
+        /// Gets the initial state.
         /// </summary>
-        /// <param name="state">The state.</param>
-        /// <param name="version">The version.</param>
-        /// <returns>
-        ///     <c>true</c> when the state contains a version; otherwise, <c>false</c>.D
-        /// </returns>
-        private static bool TryGetInitStateVersion(string state, out string version)
-        {
-            if (state.StartsWith(InitPrefix) && state.EndsWith(InitSuffix))
-            {
-                version = state.TrimStart(InitPrefix).TrimEnd(InitSuffix);
-                return true;
-            }
-
-            version = null;
-            return false;
-        }
-
-        /// <inheritdoc />
-        public override void ThrowOnUnknownInitialState(string state)
-        {
-            if (TryGetInitStateVersion(state, out var initVersion))
-            {
-                throw new InvalidOperationException(
-                    $"Version {_umbracoVersion.SemanticVersion} does not support migrating from {initVersion}."
-                    + $" Please verify which versions support migrating from {initVersion}.");
-            }
-
-            base.ThrowOnUnknownInitialState(state);
-        }
+        /// <remarks>
+        /// This equals the last migration state of v9, making that as the lowest supported version to upgrade from.
+        /// </remarks>
+        public override string InitialState => "{DED98755-4059-41BB-ADBD-3FEAB12D1D7B}"; // TODO Change this to the final v9 migration state!
```

Comment:
If we only want to support upgrading from 9.4.0 to v11, we can keep this migration as initial state:
```suggestion
        public override string InitialState => "{DED98755-4059-41BB-ADBD-3FEAB12D1D7B}"; // Final migration state of 9.4.0
```

Otherwise, only allowing v10 to v11 would require updating this to:
```suggestion
        public override string InitialState => "{B7E0D53C-2B0E-418B-AB07-2DDE486E225F}"; // Final migration state of 10.0.0
```

And then only include the migrations after 10.0.0 (currently only `AddUserGroup2LanguageTable` and `AddHasAccessToAllLanguagesColumn` to 10.2.0 and `AddBlockGridPartialViews` to 10.4.0/11.0.0).